src/OllamaApiClient.cs

```diff
@@ -355,6 +362,48 @@ private async Task EnsureSuccessStatusCode(HttpResponseMessage response)
 
 		response.EnsureSuccessStatusCode();
 	}
+	/// <summary>
+	/// Releases the resources used by the <see cref="OllamaApiClient"/> instance.
+	/// Disposes the internal HTTP client if it was created internally.
+	/// </summary>
+	public void Dispose()
+	{
+		if (_disposeHttpClient)
+			_client?.Dispose();
+	}
+
+	#region IChatClient implementation
+
+	/// <inheritdoc/>
+	ChatClientMetadata IChatClient.Metadata => new("ollama", Uri, SelectedModel);
+
+	/// <inheritdoc/>
+	async Task<ChatCompletion> IChatClient.CompleteAsync(IList<ChatMessage> chatMessages, ChatOptions? options, CancellationToken cancellationToken)
+	{
+		var request = MicrosoftAi.AbstractionMapper.ToOllamaSharpChatRequest(this, chatMessages, options, stream: false);
+		var response = await ChatAsync(request, cancellationToken).StreamToEndAsync();
+		return MicrosoftAi.AbstractionMapper.ToChatCompletion(request, response) ?? new ChatCompletion([]);
```

Comment:
Are you sure this will really work? Unless Ollama has now upgraded its streaming API to support all possible features (including function calling), I think you'll need to call Ollama's non-streaming API here. Otherwise if any tools are specified, it will return an error saying that function calling is only supported when not streaming.

If you've already verified this works with function calling then please disregard this.

Reply:
Oh maybe it does work because you're passing `stream: false`! Does that have the effect of causing it to return just a single object, and your `ProcessStreamedChatResponseAsync` interprets that equally whether it's a streamed chunk or an entire response?

Reply:
Update: yes I see you have tests for this. So I'll close this comment as resolved.

Reply:
Exactly, that' the case, yes.

Reply:
That's also what I mentioned with the `StreamToEndAsync()` method that makes methods returning `IAsyncEnumerable<T>` streamable and awaitable and even awaitable with the ability to process each chunk separately. It helped my to simplify the API a lot without having sync/async/streaming endpoints for each method overload.